src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -64,6 +64,8 @@ public function __toString(): string
      * @param string|null $key The name of the headers to return or null to get them all
      *
      * @return array<string, array<int, string|null>>|array<int, string|null>
+     *
+     * @psalm-return ($key is null ? array<string, array<int, string|null>> : array<int, string|null>)
```

Comment:
please use `@return` here as we already have the actual return type in the signature for `array`

Reply:
Sure, I updated the PR